src/actions-util.ts

```diff
@@ -821,9 +821,21 @@ export async function sendStatusReport<S extends StatusReportBase>(
   }
 }
 
+export function workflowEventName() {
+  // If the original event is dynamic CODESCANNING_EVENT_NAME will contain the right info (push/pull_request)
+  if (process.env["GITHUB_EVENT_NAME"] === "dynamic") {
+    const value = process.env["CODESCANNING_EVENT_NAME"];
+    if (value === undefined || value.length === 0) {
+      return process.env["GITHUB_EVENT_NAME"];
+    }
+    return value
```

Comment:
Minor: I think the build will fail because of this.

```suggestion
    return value;
```